cumulusci/tasks/salesforce/sourcetracking.py

```diff
@@ -0,0 +1,109 @@
+from collections import defaultdict
+import os
+
+import yaml
+
+from cumulusci.core.exceptions import TaskOptionsError
+from cumulusci.salesforce_api.metadata import ApiRetrieveUnpackaged
+from cumulusci.tasks.salesforce import BaseRetrieveMetadata
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.tasks.metadata.package import PackageXmlGenerator
+from cumulusci.tasks.metadata.package import __location__
+
+
+class ListChanges(BaseSalesforceApiTask):
+    def _run_task(self):
+        changes = self.tooling.query(
+            "SELECT MemberName, MemberType FROM SourceMember WHERE IsNameObsolete=false"
+        )
+        if changes["totalSize"]:
+            self.logger.info(
+                "Found {} changed components in the scratch org:".format(
+                    changes["totalSize"]
+                )
+            )
+            for change in changes["records"]:
+                self.logger.info(
+                    "  {}: {}".format(change["MemberType"], change["MemberName"])
+                )
+        else:
+            self.logger.info("Found no changes.")
+
+
+class RetrieveChanges(BaseRetrieveMetadata, BaseSalesforceApiTask):
+    api_class = ApiRetrieveUnpackaged
+
+    task_options = {
+        "path": {
+            "description": "The path to write the retrieved metadata",
+            "required": True,
+        },
+        "include": {
+            "description": "Include changed components matching this string.",
+            "required": True,
+        },
+        "api_version": {
+            "description": (
+                "Override the default api version for the retrieve."
+                + " Defaults to project__package__api_version"
+            )
+        },
+    }
+
+    def _init_options(self, kwargs):
+        super(RetrieveChanges, self)._init_options(kwargs)
+
+        if "api_version" not in self.options:
+            self.options[
+                "api_version"
+            ] = self.project_config.project__package__api_version
+
+        self.options["include"] = self.options["include"].split(" ")
+
+    def _get_api(self):
+        self.logger.info("Querying Salesforce for changed source members")
+        changes = self.tooling.query(
+            "SELECT MemberName, MemberType FROM SourceMember WHERE IsNameObsolete=false"
+        )
+
+        type_members = defaultdict(list)
+        for change in changes["records"]:
+            mdtype = change["MemberType"]
+            name = change["MemberName"]
+            if any(s in mdtype or s in name for s in self.options["include"]):
+                self.logger.info("Including {} ({})".format(name, mdtype))
+                type_members[mdtype].append(name)
+
+        types = []
+        for name, members in type_members.items():
```

Comment:
`types = [MetadataType(name, members) for name, members in type_members.items()]` ?

Reply:
Personally I find the non-comprehension version more readable.

Reply:
de gustibus

Reply:
Working along with black in a pre-commit hook, I suppose one could have an auto-formatter that formats things to one's personal tastes just for editing!